airflow/models/dagrun.py

```diff
@@ -228,7 +252,7 @@ def __init__(
         self.execution_date = execution_date
         self.start_date = start_date
         self.external_trigger = external_trigger
-        self.conf = conf or {}
+        self._conf = ConfDict(_conf)
```

Comment:
```suggestion
        self._conf = ConfDict(_conf or {})
```

Is this not needed to keep the current behaviour? Otherwise None would no longer be an allowed value.

Reply:
@uranusjr Fixed